### PR TITLE
Allow to skip unittests if --pdb active

### PR DESCRIPTION
**Pull request by mbyt**
One solution for #2137

### CODE REVIEW

File: _pytest/unittest.py [hunk 1]
```diff
@@ -157,9 +156,20 @@ def runtest(self):
             self._testcase(result=self)
         else:
             # disables tearDown and cleanups for post mortem debugging (see #1890)
+            # but still implements the skipping machinery (see #2137)
+            testMethod = getattr(self._testcase, self._testcase._testMethodName)
```
Comment (nicoddemus) [inline _pytest/unittest.py]: This looks correct according to the implementation in [`unittest`](https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L576), thanks!

Could you please move this entire logic to a separate method though? Something like:

```python
...
else:
    if self._handle_skip():
        return
    self._testcase.debug()
```

This makes the code easier to follow IMHO.

File: _pytest/unittest.py [hunk 2]
```diff
@@ -157,9 +156,20 @@ def runtest(self):
...
+            if (getattr(self._testcase.__class__, "__unittest_skip__", False) or
+                getattr(testMethod, "__unittest_skip__", False)):
+                # If the class or method was skipped.
+                skip_why = (getattr(self._testcase.__class__, '__unittest_skip_why__', '') or
+                            getattr(testMethod, '__unittest_skip_why__', ''))
+                try:
```
Comment (nicoddemus) [inline _pytest/unittest.py]: I recommend making an explicit version check here; try/except (specially on a generic exception like `TypeError`) can catch an exception on some unintended part of the code, making it hard to understand/debug the problem.

In short:

```python
if sys.version_info[0] == 3:
    self._testcase._addSkip(self, self._testcase, skip_why)
else:
    self._testcase._addSkip(self, skip_why)
```

**Comment by nicoddemus** [comment]
(Resolved the changelog conflict directly on the GH interface, that's neat!)

File: _pytest/unittest.py (hunk 1 above)
Comment (mbyt) [inline _pytest/unittest.py]: good idea, that makes it much easier to read. I did that in ad56cd8.

File: _pytest/unittest.py (hunk 2 above)
Comment (mbyt) [inline _pytest/unittest.py]: that was intentionally, as the python3 `_addSkip` has the same signature as the unittest2 `_addSkip` on python2, see [here](https://hg.python.org/unittest2/file/tip/unittest2/case.py#l518).

However, it was implicit. I explicitly added an comment and secured the PY2 path in 176c680.

File: _pytest/unittest.py [hunk 3]
```diff
@@ -152,14 +151,33 @@ def addSuccess(self, testcase):
...
+        # implements the skipping machinery (see #2137)
+        # analog to pythons Lib/unittest/case.py:run
+        testMethod = getattr(self._testcase, self._testcase._testMethodName)
+        if (getattr(self._testcase.__class__, "__unittest_skip__", False) or
+            getattr(testMethod, "__unittest_skip__", False)):
+            # If the class or method was skipped.
+            skip_why = (getattr(self._testcase.__class__, '__unittest_skip_why__', '') or
+                        getattr(testMethod, '__unittest_skip_why__', ''))
+            try:  # PY3, unittest2 on PY2
+                self._testcase._addSkip(self, self._testcase, skip_why)
```
Comment (nicoddemus) [inline _pytest/unittest.py]: Thanks @mbyt, now I see that it is support both `unittest` and `unittest2` on Python 2.

A try/except for TypeError still makes me uneasy, but I have no better suggestion so let's go with it. 👍

**Comment by nicoddemus** [comment]
If somebody would like more time to review this please let me know, otherwise I plan to merge this tomorrow evening.

**Comment by nicoddemus** [comment]
Thanks again @mbyt!

**Pull request merged**